ChatApp/ChatView/MessageView.swift

```diff
@@ -0,0 +1,153 @@
+        Message(text: "კარგად!", date: "მარ 14,16:10"),
+        Message(text: "ნუ მატყუებ!", date: "მარ 14,16:11")
+        
+    ]
+    
+    enum Section {
+        case main
+    }
+    
+    func setupView() {
+        backgroundColor = .white
+        setupCollectionView()
+        setupConeiner()
+    }
+    
+    func setupCollectionView() {
+        var layoutConfig = UICollectionLayoutListConfiguration(appearance: .insetGrouped)
+        layoutConfig.showsSeparators = false
+        layoutConfig.backgroundColor = .clear
+        let listLayout = UICollectionViewCompositionalLayout.list(using: layoutConfig)
+        collectionView = UICollectionView(frame: bounds, collectionViewLayout: listLayout)
+        collectionView.backgroundColor = .clear
+        self.addSubview(collectionView)
+        
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            collectionView.topAnchor.constraint(equalTo: self.topAnchor, constant: 16),
+            collectionView.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 0),
+            collectionView.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: 0),
+            //collectionView.bottomAnchor.constraint(equalTo: self.bottomAnchor, constant: -16)
+        ])
+        
+        let cellRegistration = UICollectionView.CellRegistration<MessageCell, Message> { (cell, indexPath, item) in
+            cell.label.text = item.text
+            cell.textDate.text = item.date
+            
+            //            let isMessageFromCurrentUser = item.text == "current_user_id"
+            //
+            //
+            //                cell.label.leadingAnchor.constraint(equalTo: isMessageFromCurrentUser ? cell.contentView.leadingAnchor : cell.contentView.centerXAnchor, constant: 16).isActive = true
+            //                cell.label.trailingAnchor.constraint(equalTo: isMessageFromCurrentUser ? cell.contentView.centerXAnchor : cell.contentView.trailingAnchor, constant: -16).isActive = true
+        }
+        
+        dataSource = UICollectionViewDiffableDataSource<Section, Message>(collectionView: collectionView) {
+            (collectionView: UICollectionView, indexPath: IndexPath, identifier: Message) -> UICollectionViewCell? in
+            
+            let cell = collectionView.dequeueConfiguredReusableCell(
+                using: cellRegistration,
+                for: indexPath,
+                item: identifier
+            )
+            return cell
+        }
+    }
+    
+    func applySnapshot() {
+        var snapshot: NSDiffableDataSourceSnapshot<Section, Message>!
+        
+        snapshot = NSDiffableDataSourceSnapshot<Section, Message>()
+        snapshot.appendSections([.main])
+        snapshot.appendItems(dataItems, toSection: .main)
+        
+        dataSource.apply(snapshot, animatingDifferences: true)
+    }
+    
+    func setupConeiner() {
```

Comment:
typo: setupConeiner  
+ it will be better to create separate methods instead of putting everything in one method.